core/src/main/java/io/questdb/griffin/engine/groupby/SampleByInterpolateRecordCursorFactory.java

```diff
@@ -635,5 +670,76 @@ private void nullifyRange(long lo, long hi, Record record) {
                 }
             }
         }
+
+        protected void parseParams(RecordCursor base, SqlExecutionContext executionContext) throws SqlException {
+            // factory guarantees that base cursor is not empty
+            timezoneNameFunc.init(base, executionContext);
+            offsetFunc.init(base, executionContext);
+            rules = null;
+
+            final CharSequence tz = timezoneNameFunc.getStrA(null);
+            if (tz != null) {
+                try {
+                    long opt = Timestamps.parseOffset(tz);
+                    if (opt == Long.MIN_VALUE) {
+                        // this is timezone name
+                        // fixed rules means the timezone does not have historical or daylight time changes
+                        rules = TimestampFormatUtils.EN_LOCALE.getZoneRules(
+                                Numbers.decodeLowInt(TimestampFormatUtils.EN_LOCALE.matchZone(tz, 0, tz.length())),
+                                RESOLUTION_MICROS
+                        );
+                    } else {
+                        // here timezone is in numeric offset format
+                        tzOffset = Numbers.decodeLowInt(opt) * MINUTE_MICROS;
+                        nextDstUtc = Long.MAX_VALUE;
+                    }
+                } catch (NumericException e) {
+                    throw SqlException.$(timezoneNameFuncPos, "invalid timezone: ").put(tz);
+                }
+            } else {
+                tzOffset = 0;
+                nextDstUtc = Long.MAX_VALUE;
+            }
+
+            final CharSequence offset = offsetFunc.getStrA(null);
+            if (offset != null) {
+                final long val = Timestamps.parseOffset(offset);
+                if (val == Numbers.LONG_NaN) {
+                    // bad value for offset
+                    throw SqlException.$(offsetFuncPos, "invalid offset: ").put(offset);
+                }
+                fixedOffset = Numbers.decodeLowInt(val) * MINUTE_MICROS;
+            } else {
+                fixedOffset = Long.MIN_VALUE;
+            }
+        }
+
+        private boolean areTimestampsInitialized;
+
+        protected void initTimestamps() {
+            if (areTimestampsInitialized) {
+                return;
+            }
+
+            if (!managedCursor.hasNext()) {
```

Comment:
is this branch ever taken? if the manager cursor has no record then there is an early return right from `buildMap()` => `initTimestamps()` is not called at all? or do I miss a possible path?